### ISSUE TITLE
Preview for mail with attachments

### ISSUE THREAD

**Issue by jmarceli** [fphilipe/premailer-rails#135]
I'm not sure how it's related to this gem, but when I use `skip_premailer: true` everything works fine.

Mailer function:
```ruby
  def confirmation(order, order_file)
    emails = ['someone@dot.com', 'other@mail.com']
    attachments['file.csv'] = File.read(order_file)
    mail(to: emails, subject: 'Order confirmation')
  end
```

When I send a real email it is OK but doesn't work on email preview page. I get following error:
```
NoMethodError at /rails/mailers/main_mailer/confirmation
undefined method `mime_type' for #<Array:0x00000009b58318>
```
Lines pointed in error report:
**lib/rails/templates/rails/mailers/email.html.erb**
```
</header>
<iframe seamless name="messageBody" src="?part=<%= Rack::Utils.escape(@part.mime_type) %>"></iframe>
</body>
</html>
```

I have absolutely no idea how to track down and fix this problem.

**Comment by jezstephens** [comment]
I've started having this problem after upgrading from premailer-rails 1.4.0 / premailer 1.7.3 to premailer-rails 1.8.0 / premailer 1.8.4 with rails 4.2.2.

Possibly related to rails/rails#14435?

**Comment by jezstephens** [comment]
Looks like no preview hook was registered until 1.7.0 so Premailer wasn't being used for previews. That explains why I didn't have this problem before.

**Comment by jezstephens** [comment]
Fairly sure this is a Rails bug. The problem is triggered when Premailer generates a text part, substituting the `text/html` part with a `multipart/alternative` part containing the HTML and text parts. This works OK when the original message is HTML only and the resulting multipart message is only one level deep. However, if there is further nesting due to attachments Rails does not search for the HTML part recursively as it should.

There are a couple of open PRs which aim to fix this: rails/rails#14519, rails/rails#15569.

**Comment by fphilipe** [comment]
Closing as this seems to be a rails issue.
